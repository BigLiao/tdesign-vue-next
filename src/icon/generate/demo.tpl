<template>
  <!--
    该文件有脚本 tdesign/tdesign-web-vue/src/icon/generate/index.js 自动生成。
    若需要编辑，请编辑源文件：tdesign/tdesign-web-vue/src/icon/generate/demo.tpl 。
  -->
  <div class="t-demo-iconfont">
    <p>目前设计师给到的全部图标</p><br>
    <%= icons %>
  </div>
</template>

<style lang="less">
  .t-demo-iconfont {
    .t-demo-icon {
      width: 130px;
      height: 100px;
      margin-right: 24px;
      display: inline-block;
      text-align: center;
      vertical-align: top;
      .t-icon {
        font-size: 25px;
        margin-right: 0;
        color: #999999;
      }
      .t-icon:hover {
        color: #0052d9;
        font-size: 30px;
      }
      .t-demo-icon-name {
        text-align: center;
        margin-top: 8px;
        font-size: 12px;
      }
    }
  }
</style>
