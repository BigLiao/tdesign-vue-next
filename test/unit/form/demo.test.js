import { mount } from '@vue/test-utils';
import demo from '@/examples/form/demos/base.vue';

// unit test for component in examples.
describe('Form', () => {
  it('base demo works fine', () => {
    const wrapper = mount(demo);
    expect(wrapper.element).toMatchSnapshot();
  });
});
