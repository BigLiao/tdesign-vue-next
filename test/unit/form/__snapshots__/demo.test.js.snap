// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form base demo works fine 1`] = `
<div>
  <form
    class="t-form"
  >
    <div
      class="t-form__item t-row t-form-item__name"
    >
      <div
        class="t-col t-form__label t-form__label--right t-col-1 t-form__label--colon"
      >
        <label
          for=""
        >
          姓名
        </label>
      </div>
      <div
        class="t-form__controls t-col"
      >
        <div
          class="t-form__controls--content"
        >
          <div
            class="t-input t-size-m"
          >
            <input
              class="t-input__inner"
              placeholder="请输入内容"
              type="text"
            />
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="t-form__item t-row t-form-item__tel"
    >
      <div
        class="t-col t-form__label t-form__label--right t-col-1 t-form__label--colon"
      >
        <label
          for=""
        >
          手机号
        </label>
      </div>
      <div
        class="t-form__controls t-col"
      >
        <div
          class="t-form__controls--content"
        >
          <div
            class="t-input t-size-m"
          >
            <input
              class="t-input__inner"
              placeholder="请输入内容"
              type="text"
            />
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="t-form__item t-row t-form-item__gender"
    >
      <div
        class="t-col t-form__label t-form__label--right t-col-1 t-form__label--colon"
      >
        <label
          for=""
        >
          性别
        </label>
      </div>
      <div
        class="t-form__controls t-col"
      >
        <div
          class="t-form__controls--content"
        >
          <div
            class="t-radio-group t-radio-group-solid t-radio-group-medium"
          >
            <label
              class="t-radio"
            >
              <input
                class="t-radio__former"
                name=""
                type="radio"
                value="1"
              />
              <span
                class="t-radio__input"
              />
              <span
                class="t-radio__label"
              >
                男
              </span>
            </label>
             
            <label
              class="t-radio"
            >
              <input
                class="t-radio__former"
                name=""
                type="radio"
                value="2"
              />
              <span
                class="t-radio__input"
              />
              <span
                class="t-radio__label"
              >
                女
              </span>
            </label>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="t-form__item t-row t-form-item__course"
    >
      <div
        class="t-col t-form__label t-form__label--right t-col-1 t-form__label--colon"
      >
        <label
          for=""
        >
          课程
        </label>
      </div>
      <div
        class="t-form__controls t-col"
      >
        <div
          class="t-form__controls--content"
        >
          <div
            class="t-checkbox-group"
          >
            <label
              class="t-checkbox"
            >
              <input
                class="t-checkbox__former"
                name=""
                type="checkbox"
                value="1"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              >
                语文
              </span>
            </label>
            <label
              class="t-checkbox"
            >
              <input
                class="t-checkbox__former"
                name=""
                type="checkbox"
                value="2"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              >
                数学
              </span>
            </label>
            <label
              class="t-checkbox"
            >
              <input
                class="t-checkbox__former"
                name=""
                type="checkbox"
                value="3"
              />
              <span
                class="t-checkbox__input"
              />
              <span
                class="t-checkbox__label"
              >
                英语
              </span>
            </label>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="t-form__item t-row t-form-item__status"
    >
      <div
        class="t-col t-form__label t-form__label--right t-col-1 t-form__label--colon"
      >
        <label
          for=""
        >
          状态
        </label>
      </div>
      <div
        class="t-form__controls t-col"
      >
        <div
          class="t-form__controls--content"
        >
          <div
            class="t-switch t-size-m"
          >
            <span
              class="t-switch__handle"
            />
            <div
              class="t-switch__content t-size-m"
            />
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="t-form__item t-row t-form-item__"
    >
      <div
        class="t-col t-form__label t-form__label--right t-col-1"
      >
        <label
          for=""
        />
      </div>
      <div
        class="t-form__controls t-col"
      >
        <div
          class="t-form__controls--content"
        >
          <button
            class="t-button t-size-m t-button--variant-base t-button--theme-primary"
            style="margin-right: 10px;"
            type="submit"
          >
            提交
          </button>
           
          <button
            class="t-button t-size-m t-button--variant-base t-button--theme-default"
            type="reset"
          >
            重置
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;
